projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts

```diff
@@ -171,7 +179,15 @@ export class IgxToggleDirective implements IToggleView, OnInit, OnDestroy {
     public open(overlaySettings?: OverlaySettings) {
         this._collapsed = false;
         this.cdr.detectChanges();
-        this.onOpening.emit();
+
+        const openEventArgs: CancelableEventArgs = { cancel: false };
```

Comment:
Shouldn't this code be executed before setting `this._collapsed` to `false` and trigger change detection, e.g. before doing anything else.

Reply:
No. We need this to be called after toggle is visible. This allow us to make all kind of measures, scrolling and so on by handling this event.